developer/reference/runtime/register-component.mdx

```diff
@@ -146,6 +149,67 @@ runtime.registerComponent(Cube, {
 });
 ```
 
+### Adding a description
+
+This example shows how to add an `Circle` component with a rich description. We
+use markdown formatting for our tooltip descriptions.
+
+In the builder component tray and panel, we can see an info icon next to the component label.
+
+<Frame>
+  <img
+    src="/images/register-custom-desc1.png"
+    alt="custom component info icon"
+  />
+</Frame>
+
+Hover over the label to open the description tooltip.
+
+<Frame>
+  <img
+    src="/images/register-custom-desc2.png"
+    alt="custom component description open"
+  />
+</Frame>
+
+```tsx makeswift/components.tsx
```

Comment:
Can we move the code snippet above the image. That way, we can see code first, then how it shows up in the builder?